src/main/java/com/commercetools/sync/services/ShoppingListService.java

```diff
@@ -23,51 +23,31 @@
      * <p>Note: If all the supplied keys are already cached, the cached map is returned right away with no request to
      * CTP.
      *
-     * @param keys the shopping list keys to fetch and cache the ids for.
+     * @param shoppingListKeys the shopping list keys to fetch and cache the ids for.
      *
      * @return {@link CompletionStage}&lt;{@link Map}&gt; in which the result of it's completion contains a map of all
      *          shopping list keys -&gt; ids
      */
     @Nonnull
-    CompletionStage<Map<String, String>> cacheKeysToIds(@Nonnull Set<String> keys);
+    CompletionStage<Map<String, String>> cacheKeysToIds(@Nonnull Set<String> shoppingListKeys);
 
     /**
-     * Given a {@code key}, this method first checks if a cached map of shopping list keys -&gt; ids is not empty.
-     * If not, it returns a completed future that contains an optional that contains what this key maps to in
-     * the cache. If the cache is empty, the method populates the cache with the mapping of all shopping list keys
-     * to ids in the CTP project, by querying the CTP project for all shopping lists.
+     * Given a {@link Set} of shopping list keys, this method fetches a set of all the shopping lists, matching given
+     * set of keys in the CTP project, defined in an injected {@link SphereClient}. A mapping of the key to the id
+     * of the fetched shopping lists is persisted in an in-memory map.
      *
-     * <p>After that, the method returns a {@link CompletionStage}&lt;{@link Optional}&lt;{@link String}&gt;&gt;
```

Comment:
You might need to check merge conflicts in this class, I assumed this branch has the correct methods, but I am not sure right now, as a reviewer, I am having a hard time to understand which code should be in which branch 🤔